src/lib.rs

```diff
@@ -37,11 +35,33 @@ pub fn init() -> Result<(), SetLoggerError> {
     log::set_logger(&EguiLogger).map(|()| log::set_max_level(log::LevelFilter::Info))
 }
 
-static LOG: Mutex<Vec<(log::Level, String)>> = Mutex::new(Vec::new());
+type GlobalLog = Vec<(log::Level, String)>;
+
+static LOG: Mutex<GlobalLog> = Mutex::new(Vec::new());
 
 static LOGGER_UI: once_cell::sync::Lazy<Mutex<LoggerUi>> =
     once_cell::sync::Lazy::new(Default::default);
 
+fn try_mut_log<F, T>(f: F) -> Option<T>
+where
+    F: FnOnce(&mut GlobalLog) -> T,
+{
+    match LOG.lock() {
+        Ok(ref mut global_log) => Some((f)(global_log)),
+        Err(_) => None,
+    }
+}
+
+fn try_get_log<F, T>(f: F) -> Option<T>
+where
+    F: FnOnce(&GlobalLog) -> T,
+{
+    match LOG.lock() {
+        Ok(ref global_log) => Some((f)(global_log)),
+        Err(_) => None,
+    }
+}
+
```

Comment:
This looks like a good idea. I think that this will make thinks easier and saver in the future.